src/components/logger.ts

```diff
@@ -23,6 +23,13 @@ export class Logger {
         }
     }
 
+    logCommand(message: string, command: string, args: string[] | undefined) {
+        this.addLogMessage(message + ': ' + command)
+        if (args) {
+            this.addLogMessage(message + ' args: ' + JSON.stringify(args))
+        }
+    }
```

Comment:
We had better always add arguments to logs as follows:

```ts
    logCommand(message: string, command: string, args: string[] = []) {
        this.addLogMessage(message + ': ' + command)
        this.addLogMessage(message + ' args: ' + JSON.stringify(args))
    }
```

It makes clear that there is no argument if it is empty, and it makes reading log messages much easier.